pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -521,6 +521,10 @@ def get_min_variation(ch_sample: ChannelSamples) -> int:
                 )
                 / 1000
             )
+        if "nsteps" not in options:
+            options["nsteps"] = max(
+                1000, self._tot_duration // options["max_step"]
+            )
```

Comment:
I'm not sure `nsteps` refers to the total number of steps over the entire sequence. From the [Qutip docs](https://qutip.org/docs/latest/guide/dynamics/dynamics-options.html#setting-options-for-the-dynamics-solvers) and the define it as: "Max. number of steps to take for each interval"

What is an interval though?

Reply:
Currently, this would be failing (with the error asking to raise nsteps) if the user was computing a sequence with one Pulse composed of a customwaveform of duration > 1000 changing value over large distance at each ns (for instance oscillating between 0 and 1) (untested, but I think this is the case)

Reply:
Tested: Fails
```
import pulser
import pulser_simulation
seq = pulser.Sequence(pulser.Register.square(2, 5), pulser.AnalogDevice)
seq.declare_channel("ryd_glob", "rydberg_global")
oscillations = [0, 1] * 1000
seq.add(
    pulser.Pulse.ConstantDetuning(
        pulser.CustomWaveform([0, 1] * 1000),
        0,
        0,
    ),
    channel="ryd_glob",
)
seq.draw()
sim = pulser_simulation.QutipEmulator.from_sequence(seq, evaluation_times="Minimal")
sim.run().get_final_state()
```
Exception: ODE integration error: Try to increase the allowed number of substeps by increasing the nsteps parameter in the Options class.

Reply:
Ok, but this is an unrealistic edge case so I wouldn't worry about. Plus, the user can still manually increase `nsteps` to fix it.

Reply:
But it's still a nice feature, right ? It does not take too much effort on our side to introduce this

Reply:
Yes, let's keep it :)